tests/unit_tests/client/address_book/impl.cc

```diff
@@ -47,36 +47,75 @@ struct SubscriptionFixture {
 
   /// @brief Validates given lines as proven addressbook host/address pairs
   /// @param lines Lines to validate
-  /// @return Only valid data that was parsed
-  auto Validate(const std::vector<std::string>& lines) {
+  /// @return True if all subscription lines pass validation
+  bool Validate(const std::vector<std::string>& lines) {
     std::stringstream stream;
-    for (auto const& l : lines)
+    for (const auto& l : lines)
       stream << l << "\n";
-    return book.ValidateSubscription(stream);
+    const auto validated = book.ValidateSubscription(stream);
+    return validated.size() == lines.size();
   }
 
-  /// @brief Compares and validates lines against subscription fixture
-  /// @return False if test data does not match subscription fixture
-  bool Validate() {
-    // Save fixture (malformed or redundant lines will be removed)
-    for (auto const& s : subscription) {
-      const std::string valid(s);
-      lines.push_back(valid);
-    }
-    // Validate
-    std::vector<std::string> validated;
-    for (auto const& v : Validate(lines))
-      validated.push_back(v.first + "=" + v.second.ToBase64());
-    return validated != subscription ? false : true;
+  /// @brief Appends lines to fixture subscription
+  /// @param new_entries Entries to add to the test subscription
+  /// @param duplicate Boolean flag to append unique or duplicate subscriptions
+  /// @return Test subscription lines
+  const std::vector<std::string> CreateTestSubscription(
+      const std::vector<std::string> new_entries = {""},
+      const bool duplicate = false)
+  {
+    std::vector<std::string> new_lines {""};
+    for (const auto& n : new_entries)
+      new_lines.push_back(n);
+    if (duplicate)
+      {
+        for (const auto& s : subscription_duplicate)
+          new_lines.push_back(s);
+      }
+    else
+      {
+        for (const auto& u : subscription_unique)
+          new_lines.push_back(u);
+      }
+    return new_lines;
   }
 
-  /// @brief Subscription with valid hosts
-  const std::vector<std::string> subscription {{
+  /// @brief Load fixture subscription addresses into memory
+  /// @param load_all Boolean flag to load both unique and duplicate addresses
+  /// @notes Mocks addresses being loaded from hosts.txt
+  void LoadAddresses(const bool load_all = false)
+  {
+    std::vector<std::string> addresses {""};
+    if (load_all)
+      addresses = CreateTestSubscription(subscription_duplicate);
+    else
+      addresses = subscription_duplicate;
+    for (const auto& a : addresses)
+      {
+        std::vector<std::string> ident_hash {""};
+        boost::split(ident_hash, a, boost::is_any_of("="));
+        const std::string addr = ident_hash[0], 
+                          ident_str = ident_hash[1];
+        kovri::core::IdentityEx ident;
+        ident.FromBase64(ident_str);
+        book.InsertAddress(addr, ident);
+      }
+  }
```

Comment:
I know your comment was from a previous commit, but I would still like your input on `GetAddressBookEntry` and `AddressBookEntry`.

I would like to add `AddressBookEntry` to the impl proper, think it would be useful for an API rewrite.

For example, we could pass around individual entries as const `AddressBookEntry` references, keeping the internal representation private.

This approach would also allow us to store other metadata (thinking of making life easier for #385).

What do you think?